### PR TITLE
[#1805] Add logic to EntityViewManager and ProxyFactory to maintain e…

### LINKED ISSUE #1805: ProxyFactory#getProxy seems to re-use EVM instances from the cache without properly settings optionalParameters
Assuming that one uses `entityViewConfiguration.addOptionalParameters` to insert default parameters for an EVM to carry (specifically beans on needs to use during Pre/Post remove listeners), it seems that during the invocation attempt for thus listeners, and EVM instance is created via `context.getEntityViewManager().getSerializableDelegate(view.$$_getEntityViewClass())`
 which ultimately will use the `ProxyFactory` to create or **reuse* an EVM.

When an EVM is "reused" it is not ensured, that this EVM has the optionalParameters applied and thus we can end up having an EVM without those parameters included.

### PR DESCRIPTION
**Pull request by Mobe91**
Fixes #1805

### CODE REVIEW

File: entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/EntityViewManagerImpl.java [hunk 1]
```diff
@@ -1450,6 +1450,11 @@ public EntityViewManager getSerializableDelegate(Class<?> entityViewClass) {
         return serializableDelegates.get(proxyFactory.getProxy(this, metamodel.managedViewOrError(entityViewClass)));
     }
 
+    @Override
+    public void close() {
+        proxyFactory.clear();
```
Comment (beikov) [inline entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/EntityViewManagerImpl.java:1455]: I think we should add a `static volatile boolean closed` as well and set this here, as well as check on every public method if it is closed to throw an exception when that is the case.
Apart from that, do you know if the static implementations (generated by the annotation processor) are "unloaded" properly as well?

**Comment by EugenMayer** [comment]
Would be great to have this on the main branch to be able to test it all together!

File: entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/EntityViewManagerImpl.java (hunk 1 above)
Comment (Mobe91) [reply entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/EntityViewManagerImpl.java:1455]: > I think we should add a static volatile boolean closed as well and set this here, as well as check on every public method if it is closed to throw an exception when that is the case.

Why would we do that? What's the point? The contract of the close method clearly states that any invocations on an entity view manager that has been closed is undefined.

> Apart from that, do you know if the static implementations (generated by the annotation processor) are "unloaded" properly as well?

No, I need to check that.

Comment (beikov) [reply entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/EntityViewManagerImpl.java:1455]: >Why would we do that? What's the point? The contract of the close method clearly states that any invocations on an entity view manager that has been closed is undefined.

Yes, but undefined is not very user friendly. If we get into some state when bad things might happen after closing, then we should ensure a user is not using the EVM anymore after a close. I created https://github.com/Blazebit/blaze-persistence/issues/1813 for this task as I don't want to hold back the release for this any further than necessary, but if you find time to work on that, I'd very much appreciate it.

Comment (Mobe91) [reply entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/EntityViewManagerImpl.java:1455]: > Apart from that, do you know if the static implementations (generated by the annotation processor) are "unloaded" properly as well?

Judging from `com.blazebit.persistence.view.impl.proxy.ProxyFactory#loadImplementation` static implementations should be handled properly as well by this change.

**Pull request merged**